test/auth_test.rb

```diff
@@ -119,6 +119,55 @@ def app
         assert_equal "https://sso.heroku.com/initech",
                      last_response.headers["Location"]
       end
+
+      describe "with a valid token" do
+        let(:rack_env) do
+          {
+            "rack.session" => {
+              "sso_entity" => "initech",
+              "access_token" => "1234",
+              "access_token_expires_at" => (Time.now.getlocal + 3600).to_s
+            }
+          }
+        end
+
+        it "authorizes when i have a valid access token" do
+          stub_heroku_api do
+            get("/account") { status(200) }
+          end
+
+          post "/oauth/authorize", { client_id: "dashboard" }, rack_env
+
+          assert_equal 302, last_response.status
+          assert_equal "https://dashboard.heroku.com/oauth/callback/heroku" +
+                       "?code=454118bc-902d-4a2c-9d5b-e2a2abb91f6e",
```

Comment:
As someone new to identity, the expected valued of this assertion is difficult to understand as it required combing through `identity/auth.rb`, `service_stubs.rb', and `heroku_api_stub.rb`. 

It would be easier for future developers if this assertion were extracted into a custom assertion such as `assert_response_redirects_with_oauth_callback` written something like this:

```ruby
def assert_response_redirects_with_oauth_callback
  assert_equal 302, last_response.status
  assert_equal "#{HerokuAPIStub::AUTHORIZATION[:client][:redirect_uri]}?code=#{HerokuAPIStub::AUTHORIZATION[:grant][:code]}", last_response.headers["Location"]
```

And extract the `authorization` from `HerokuAPIStub`'s `POST /oauth/authorizations` into the `HerokuAPIStub::AUTHORIZATION` constant.